sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -151,10 +169,69 @@ final class OffsetWindowFunctionFrame(
     }
     inputIndex += 1
```

Comment:
here we reset the `inputIndex`. Does it make sense for nth_value?

Reply:
`nth_value` not use `FrameLessOffsetWindowFunctionFrame`.